packages/entity-server/src/routes/hierarchy/EntitySearchRoute.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { Route } from '@tupaia/server-boilerplate';
+import { QUERY_CONJUNCTIONS } from '@tupaia/database';
+import { formatEntitiesForResponse } from './format';
+import {
+  MultiEntityRequest,
+  MultiEntityRequestParams,
+  MultiEntityRequestBody,
+  EntityRequestQuery,
+  EntityResponse,
+} from './types';
+
+const { RAW } = QUERY_CONJUNCTIONS;
+
+export type EntitySearchRequest = MultiEntityRequest<
+  MultiEntityRequestParams & { searchString: string },
+  EntityResponse[],
+  MultiEntityRequestBody,
+  EntityRequestQuery & { pageSize?: number; page?: number }
+>;
+export class EntitySearchRoute extends Route<EntitySearchRequest> {
```

Comment:
I was really wanting this to be something we could support using just the existing routes. We already have the ability to filter results by a search term, so if we added the ability to paginate and sort across all routes then we'd almost be there.

But I think the custom sort that we do here is quite hard to support. Even if we broke the request from the `tupaia-web-server` into multiple api calls, getting the pagination working across those multiple calls would prove difficult.

So happy with this being added, and it'll probably get nice re-use on other platforms too 👍

Reply:
This is almost exactly the thought process I went through too. Tried supporting through existing routes with multiple calls, found that difficult for pagination, figured this route would be useful for other platforms 😅

Reply:
Thanks for thinking that through, it's always nice to try and keep the api lean. 👍 

I think if we returned a bit of metadata with the request, we could look at supporting this with the existing routes.
eg. something like:
```
/hierarchy/explore/explore/descendants?field=name&filter=name=@cat&page=0&pageSize=10&sort=name asc
=>
{
  page: 0,
  totalPages: 10,
  data: [
    "Yacata Nursing Station",
    "Lemakot Catholic Miss",
    ...
  ]
}
```

Honestly, that makes for a more powerful api generally I think, and would be a nice change to make at some point. But for now I'm happy with going with what we've got 👍